src/course.py

```diff
@@ -0,0 +1,92 @@
+import os
+import re
+import urllib.parse
+
+from bs4 import BeautifulSoup
+
+import globals
+from resource import Resource
+
+
+class Course:
+    def __init__(self, course_name, course_url):
+        self.name = course_name
+        self.course_url = course_url
```

Comment:
`self.course_url` doesn't seem to be needed anywhere else other than once to generate the course page, one line below it. Maybe having it as a class variable is a bit unnecessary.